sound/soc/sof/ipc4-topology.c

```diff
@@ -1061,15 +1061,117 @@ static bool sof_ipc4_is_single_format(struct snd_sof_dev *sdev,
 	return true;
 }
 
+static bool sof_ipc4_copier_is_single_format(struct snd_sof_dev *sdev,
+					     struct sof_ipc4_pin_format *pin_fmts,
+					     u32 pin_fmts_size)
```

Comment:
commit message

````
When a host copier is connected to an SRC, it will need to support
multiple sample rates in its input formats.
````

This is true only for playback

````
If there's a mixin/mixout in
the path along with an SRC,
````

How would the copier know this?

It's unclear what this patch is doing just by looking at the explanations here.